torch/utils/_foreach_utils.py

```diff
@@ -64,6 +90,6 @@ def _group_tensors_by_device_and_dtype(tensorlistlist: List[List[Tensor]],
     return per_device_and_dtype_tensors
 
 def _has_foreach_support(tensors: List[Tensor], device: torch.device) -> bool:
-    if (device.type != "cpu" and device.type != SupportForeachDeviceType._default_device_type) or torch.jit.is_scripting():
+    if device.type not in SupportForeachDevices.get_device_types_and("cpu") or torch.jit.is_scripting():
```

Comment:
Ah, if this is the only usage, maybe we should just add cpu to the SupportForeachDevices list?